prog/hints.go

```diff
@@ -84,9 +84,86 @@ func generateHints(p *Prog, compMap CompMap, c *Call, arg Arg, exec func(newP *P
 }
 
 func checkConstArg(arg *ConstArg, compMap CompMap, cb func(newArg Arg)) {
-	for v, _ := range compMap[arg.Val] {
-		cb(constArg(arg.typ, v))
+	replacers := make(uint64Set)
+	for _, mutation := range mutations {
+		mutation(arg.Val, compMap, replacers)
+	}
+	for replacer, _ := range replacers {
+		// Filter out "special" values. The fuzzer will try them anyways.
+		if ok, _ := specialIntsSet[replacer]; !ok {
```

Comment:
specialIntsSet should be checked inside of shrinkExpand before we merge old and new values. Consider that we are replacing with a 4-byte 0, but high part of the original value is 0x12345678, replacer with be 0x1234567800000000, which will not match specialIntsSet.